draft-ietf-tls-dtls-connection-id.md

```diff
@@ -229,22 +229,41 @@ zeroes
 {:br}
 
 In addition, the CID value is included in the MAC calculation for the
-DTLS record as shown below. The MAC algorithm described in Section
-4.1.2.1 of {{RFC6347}} and Section 6.2.3.1 of {{RFC5246}} is extended
+DTLS record layer. At the time of writing ciphers using authenticated 
+encryption with additional data (AEAD) were state-of-the-art. Hence, this 
+specification updates only the additional data calculation defined in 
+Section 6.2.3.3 of {{RFC5246}}, which is re-used by Section
+4.1.2.1 of {{RFC6347}}. 
+
+The additional data calculation is extended
 as follows:
 
 ~~~~
-      MAC(MAC_write_key, DTLSCompressed.epoch +
-                            DTLSCompressed.sequence_number +
-                            tls12_cid +
-                            DTLSCompressed.version +
-                            cid_length +        // New input
-                            cid +               // New input
-                            DTLSWrappedCompressed.length +
-                            DTLSWrappedCompressed.fragment);
-   where "+" denotes concatenation.
+    additional_data = seq_num + DTLSWrappedCompressed.type + 
+	                  DTLSCompressed.version + DTLSCompressed.length + 
+					  cid + cid_length + cid_codepoint + 
+					  DTLSWrappedCompressed.length + 
+					  DTLSWrappedCompressed.fragment;
+					  
+    where "+" denotes concatenation. 
```

Comment:
I modified the list of fields in the additional data calculation.

Reply:
The tls `seq_num` of https://tools.ietf.org/html/rfc5246#section-6.2.3.3 is in my opinion not the dtls `seq_num` of https://tools.ietf.org/html/rfc6347#section-4.1 ! Please don't remove the `epoch` field.

Reply:
Uups, sorry forget the upper comment about the epoch! I read the last commit in isolation ;-(.

Reply:
Thanks Hannes, looks good to me.

PS: not that it makes any difference, but adding the fields in the order in which they appear in the header
```
   seq_num + type + version + cid_length + cid + length
```
looks slightly more consistent with the previously established convention?

Reply:
Another small note: I think we should not remove the MAC computation.  I understand we want to move to an AEAD-only world, but this needs to work with every mode supported in 1.2...

Reply:
> PS: not that it makes any difference, but adding the fields in the order in which they appear in the header

There is no field `cid_length` in the header :-).

Reply:
While there is no such field in the header that doesn't mean we cannot include it in the additional data calculation.

Reply:
Thomas, if you can add the text for the other MAC calculations then that's fine with me. I only care about the AEAD ciphers since we recommend them in the DTLS / TLS IoT profiles draft.

Reply:
> While there is no such field in the header that doesn't mean we cannot include it in the additional data calculation.

Sure, but why should it then be added?

Reply:
Which size is considered for the `cid_length`? The same as in the extension (`opaque cid<0..2^8-1>`, so 1 byte)?

Reply:
It would be 1 byte length but your remark is fair. IMHO one would have to do a formal analysis to say whether it really has to be added or not. Currently, we are just adding fields to the additional data / MAC calculation without much justification

Reply:
OK, thanks! That helps me to understand the current definition.